### ISSUE TITLE
Phase 8 polish: tighten superior_force_usually_wins threshold from 180/200

### ISSUE THREAD

**Issue by Troll-Phace** [Troll-Phace/aegis#11]
## Context

Surfaced by the Phase 8 code review. The `superior_force_usually_wins` property test asserts blue wins ≥180 of 200 runs when blue has a 2:1 strength advantage with equal lethality. Given the math, the realistic expectation is 200/200 — the threshold has so much slack that a real regression could pass silently.

## Where

- `src-tauri/tests/combat_properties.rs:79-94`:

```rust
#[test]
fn superior_force_usually_wins() {
    let runs = 200;
    let mut blue_wins = 0;
    for i in 0..runs {
        let (mut w, bid, rid) = build_duel(i as u64, 100.0, 50.0, 0.05);
        let (bs, rs, _) = run_to_resolution(&mut w, bid, rid);
        if bs > 0.0 && rs <= 0.0 {
            blue_wins += 1;
        }
    }
    assert!(
        blue_wins >= 180,
        \"expected blue (2:1 advantage) to win >= 180/200, got {blue_wins}\"
    );
}
```

## Why

With `blue.strength=100`, `red.strength=50`, `lethality=0.05` on both sides, and jitter `∈ [0.7, 1.3)`:

- Per-tick red_loss = `100 * 0.05 * dt * jitter_red ≈ 5 * jitter_red` per second
- Per-tick blue_loss = `50 * 0.05 * dt * jitter_blue ≈ 2.5 * jitter_blue` per second

Blue always outpaces red by a factor of ~2, regardless of the jitter draw. The Lanchester square law says blue wins every time (blue finishes with ≈ `sqrt(100² − 50²) = sqrt(7500) ≈ 86.6` strength). The test's expected value is 200/200, not \"at least 180.\" The 180 threshold has a 10% regression tolerance — enough to hide a real bug that makes blue lose ~5% of the time.

## Acceptance criteria

Pick one:

- **Option A** (recommended) — tighten to ≥198/200 or ≥199/200 and add a comment explaining the 1-2 run slack is rounding/float headroom, not a stochastic tolerance
- **Option B** — replace with stronger assertions:
  1. Blue wins 200/200
  2. Mean `final_blue_strength` across runs is within `±5%` of the Lanchester closed-form prediction (≈86.6)
  3. Std-dev is non-zero (i.e., jitter is actually doing something)

Option B is the ideal — it catches both \"blue stopped winning\" regressions AND \"jitter broke and became deterministic\" regressions.

- [ ] Threshold tightened per chosen option
- [ ] Test still passes (run it ≥5 times to verify stability)
- [ ] Assertion message documents the rationale

## Delegate to

`test-engineer`

## References

- Phase 8 code review, \"polish / structure before Phase 16\" item 11
- `docs/ARCHITECTURE.md` §4.4 — Lanchester square law
